File: test_speed.cpp
#include <iostream>
#include <chrono>

int main() {
    // Фіксуємо час старту
    auto start = std::chrono::high_resolution_clock::now();

    long long total = 0;
    for (int i = 0; i < 100000000; ++i) { // 100 мільйонів ітерацій
        total += i;
    }

    // Фіксуємо час фінішу
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = end - start;

    std::cout << "C++ Result: " << total << std::endl;
    std::cout << "C++ Time: " << elapsed.count() << " seconds" << std::endl;

    return 0;
}

File: test_speed.rs
use std::time::Instant;

fn main() {
    // Фіксуємо час старту
    let start = Instant::now();

    let mut total: u64 = 0;
    for i in 0..100_000_000 { // 100 мільйонів ітерацій
        total += i;
    }

    // Фіксуємо час фінішу
    let duration = start.elapsed();

    println!("Rust Result: {}", total);
    println!("Rust Time: {:?}", duration);
}